### ISSUE TITLE
Handle invalid feature form values (unify constraints)

### ISSUE THREAD

**Issue by tomasMizera** [lutraconsulting/input#1507]
It would be great to change a little bit how we handle user input in forms. It is not the greatest experience to just remove data from field when they are invalid, e.g. when someone scans QR code with text into numeric field or types in `34,51hg` to numeric field. 

**Right now when user saves the form with invalid value, last valid value is saved.** This can be misleading, user thinks value A is saved, but actually some other value is saved. 

Pr #1500 introduced new way how to handle invalid values. There is a text under field name (previously used only for constraint description) that can show texts like _"Not a valid number"_ or _"Number out of range"_ for numeric fields. We can easily use this for all other fields.

Proposal:
 - Even when value is invalid, save it to `mFeatureLayerPair` in `AttributeController`
 - Let's have a boolean array (flags) with entry for each field to know if value in a field is valid
 - QGIS Constraints and invalid values could be unified:
    - Controller no longer serving hard and soft constraints, rather `errorMessage` for each field and `errorLevel (info, warning, error)`
    - Invalid value in field and hard constraint would be `error`
    - Soft constraints as `warnings`
 - Do not let user save form if there is any field with `error` state
 - `errorMessage` is set in `AttributeController` and shown under field name with color according to `errorLevel`

**TLDR** Let user type in any value (even invalid) to field and tell him it is an invalid value (and why). Do not let them save the form when there is an invalid value.

See more in #1500 comments

**Comment by tomasMizera** [comment]
Also add description when `not null` and `unique` constraints are applied. Currently there is no description for these constraints

**Comment by tomasMizera** [comment]
Should also deal with https://github.com/lutraconsulting/input/pull/1500#discussion_r655933272
